### PR TITLE
Release notes for 2.24.0.

### PR DESCRIPTION
**Pull request by rstam**

### CODE REVIEW

**Review by damieng — COMMENTED** [review]
No mention of nested AsQueryable support?

**Comment by rstam** [comment]
> No mention of nested AsQueryable support?

Good question. I didn't want to mention it since no user should ever be writing queries this way. This feature was solely for the benefit of the EF Core Provider.

@JamesKovacs what do you think?

**Comment by JamesKovacs** [comment]
> > No mention of nested AsQueryable support?
> 
> Good question. I didn't want to mention it since no user should ever be writing queries this way. This feature was solely for the benefit of the EF Core Provider.
> 
> @JamesKovacs what do you think?

We should mention it on the off chance that it affects a user's query. We may want to mention that it is intended for internal use only (or however we want to word it).

File: Release Notes/Release Notes v2.24.0.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+This is the general availability release for the 2.24.0 version of the driver.
+
+The main new features in 2.24.0 include:
+
+* New DistinctMany method in IMongoCollection
+* Support for latest $dateFromString optional arguments
+* ExceededTimeLimit server error is now retryable for reads as well as writes
+* Support sort by score in $search stage
+* New custom method to test if a field exists or is missing in LINQ
```
Comment (JamesKovacs) [inline Release Notes/Release Notes v2.24.0.md:11]: >New custom methods to test if a field exists or is missing in LINQ (Mql.IsMissing, Mql.Exists, Mql.IsNullOrMissing)

**Review by JamesKovacs — CHANGES_REQUESTED** [review]
As noted in my earlier comment, I would mention nested AsQueryable support. I would also add:
- Updated authentication to occur over OP_MSG on supporting servers to improve MongoDB Atlas Serverless compatibility
- Use polling monitoring when running within a FaaS environment such as AWS Lambda
- Fixed segfault in Kerberos (libgssapi) on newer Linux distros

File: Release Notes/Release Notes v2.24.0.md (hunk 1 above)
Comment (rstam) [reply Release Notes/Release Notes v2.24.0.md]: Done

**Review by JamesKovacs — APPROVED** [review]
LGTM

**Comment by Evengard** [comment]
The release seems to be absent on NuGet and Github Releases

**Comment by rstam** [comment]
This is in preparation of the upcoming release of 2.24.0. The release has not happened yet.

**Pull request merged**